### ISSUE TITLE
[bug] `v1.15` executes tasks with `runInCI:false` that did not run under `v1.14`

### ISSUE THREAD

**Issue by mrobinsn** [moonrepo/moon#1153]
This was hard to track down, but the behavior is that some tasks that I have marked with `runInCI: false` are now unexpectedly running during `moon ci` now that I've upgraded to moon `v1.15`. 

This is a repro that I've put together, the issue seems to stem from cross-project dependencies.

`.moon/workspace.yml`:
```yaml
# https://moonrepo.dev/docs/config/workspace
$schema: 'https://moonrepo.dev/schemas/workspace.json'

versionConstraint: '>=1.12.1'
runner:
  logRunningCommand: false
telemetry: false
projects:
  sources:
    project1: '.'
    project2: 'project2'
  globs: []
vcs:
  manager: git
  defaultBranch: main
```

`moon.yml`:
```yaml
$schema: https://moonrepo.dev/schemas/project.json
language: go
tasks:
  create-dir:
    command: mkdir -p @out(0)
    inputs: []
    outputs:
    - my-dir/
  create-file:
    # this task is set up to fail so its obvious when it ran during moon ci
    command: touch @out(0) && exit 1
    inputs: []
    outputs:
    - my-dir/test.txt
    deps:
    - create-dir
    - project2:install-tool
    options:
        runInCI: false
  create-other-file:
    command: touch @out(0)
    inputs: []
    outputs:
    - my-dir/test-other.txt
    deps:
    - create-dir
```

`project2/moon.yml`:
```yaml
$schema: https://moonrepo.dev/schemas/project.json
language: go
tasks:
  install-tool:
    command: touch $workspaceRoot/some-tool
    inputs: []
    outputs:
    - /some-tool
```

And then run `moon ci`

With moon `v1.14.X`:
```bash
❯ moon ci
▪▪▪▪ Gathering touched files
  .gitignore
  .moon/workspace.yml
  moon.yml
▪▪▪▪ Gathering runnable targets
  project1:create-dir
  project1:create-other-file
  project2:install-tool
▪▪▪▪ Generating dependency graph
Target count: 3
Action count: 9
▪▪▪▪ Running targets
▪▪▪▪ project1:create-dir
▪▪▪▪ project1:create-dir (8ms, c89a7f22)
▪▪▪▪ project2:install-tool
▪▪▪▪ project1:create-other-file
▪▪▪▪ project2:install-tool (6ms, 1878be23)
▪▪▪▪ project1:create-other-file (6ms, 22439ce6)
▪▪▪▪ Summary

No failed actions to summarize.

▪▪▪▪ Stats

skip SetupSystemTool (skipped)
pass SyncSystemProject(project2) (0s)
skip InstallSystemDepsInProject(project1) (skipped)
pass SyncSystemProject(project1) (0s)
pass RunTarget(project1:create-dir) (20ms)
skip InstallSystemDepsInProject(project2) (skipped)
pass SyncWorkspace (0s)
pass RunTarget(project2:install-tool) (17ms)
pass RunTarget(project1:create-other-file) (17ms)


Actions: 6 completed, 3 skipped
   Time: 38ms
```

With moon `v1.15.X`:
```bash
❯ moon ci
▪▪▪▪ Gathering touched files
  .gitignore
  .moon/workspace.yml
  moon.yml
▪▪▪▪ Gathering runnable targets
  project1:create-dir
  project1:create-other-file
  project2:install-tool
▪▪▪▪ Generating action graph
Target count: 3
Action count: 8
▪▪▪▪ Running targets
▪▪▪▪ project2:install-tool
▪▪▪▪ project2:install-tool (4ms, 1878be23)
▪▪▪▪ project1:create-dir
▪▪▪▪ project1:create-dir (4ms, 8d82da93)
▪▪▪▪ project1:create-other-file
▪▪▪▪ project1:create-file
▪▪▪▪ project1:create-other-file (5ms, c6e0de75)
▪▪▪▪ project1:create-file (5ms, 3ec4bdc3)
▪▪▪▪ Summary

▪▪▪▪ project1:create-file


▪▪▪▪ Stats

pass SyncWorkspace (0s)
skip SetupSystemTool (skipped)
pass SyncSystemProject(project2) (0s)
pass SyncSystemProject(project1) (0s)
pass RunTask(project2:install-tool) (21ms)
pass RunTask(project1:create-dir) (7ms)
fail RunTask(project1:create-file) (11ms)
pass RunTask(project1:create-other-file) (11ms)


Actions: 6 completed, 1 failed, 1 skipped
   Time: 41ms
```

If this is the new intended behavior, would it be possible to have a "no I really mean it" version of `runInCI: false`?
My use case is that I have expensive IaC integration tests that I never want to run as part of `moon ci`, they are intended to always be executed with purpose via `moon run <test-name>`.

**Comment by milesj** [comment]
@mrobinsn This is a bit weird since we explicitly check it here: https://github.com/moonrepo/moon/blob/master/crates/cli/src/commands/ci.rs#L106

A shot in the dark, but I noticed there's extra spacing before `runInCI: false`. If you fix that does it go away? I know yaml is strict about this kind of stuff...

**Comment by milesj** [comment]
I think I found the problem, added a fix here: https://github.com/moonrepo/moon/pull/1155

**Comment by milesj** [comment]
Ok can you try 1.16 and see if that fixed it.

**Comment by mrobinsn** [comment]
Same issue unfortunately 😢 

I double checked the yaml formatting, it's good, just came across a little weird in the copy paste to Github. 

Debug logs on the repro do show:
```
[DEBUG 10:58:58] moon_cli::commands::ci:ci  Not running target project1:create-file because it either has no outputs or runInCI is false
```
But then it still runs it anyway

**Comment by milesj** [comment]
What... ok let me dig further...

**Comment by milesj** [comment]
@mrobinsn Seems to work for me?

1.15:
```
$ CI=true moon ci
▪▪▪▪ Gathering touched files
▪▪▪▪ Gathering runnable targets
  project1:create-dir
  project1:create-other-file
  project2:install-tool
▪▪▪▪ Generating action graph
Target count: 3
Action count: 8
▪▪▪▪ Running targets
▪▪▪▪ project2:install-tool
▪▪▪▪ project2:install-tool (9ms, ba876655)
▪▪▪▪ project1:create-dir
▪▪▪▪ project1:create-dir (5ms, 4d1c65f4)
▪▪▪▪ project1:create-other-file
▪▪▪▪ project1:create-other-file (4ms, c64f0bad)
▪▪▪▪ project1:create-file
▪▪▪▪ project1:create-file (4ms, 74bd3ef0)
▪▪▪▪ Summary

▪▪▪▪ project1:create-file


▪▪▪▪ Stats

pass SyncWorkspace (0s)
skip SetupSystemTool (skipped)
pass SyncSystemProject(project2) (0s)
pass SyncSystemProject(project1) (0s)
pass RunTask(project2:install-tool) (23ms)
pass RunTask(project1:create-dir) (8ms)
fail RunTask(project1:create-file) (14ms)
pass RunTask(project1:create-other-file) (7ms)


Actions: 6 completed, 1 failed, 1 skipped
   Time: 48ms
```

1.16:
```
$ CI=true moon ci
▪▪▪▪ Gathering touched files
▪▪▪▪ Gathering runnable targets
  project1:create-dir
  project1:create-other-file
  project2:install-tool
▪▪▪▪ Generating action graph
Target count: 3
Action count: 7
▪▪▪▪ Running targets
▪▪▪▪ project2:install-tool (cached from previous run, ba876655)
▪▪▪▪ project1:create-dir (cached from previous run, 4d1c65f4)
▪▪▪▪ project1:create-other-file (cached from previous run, c64f0bad)
▪▪▪▪ Summary

No failed actions to summarize.

▪▪▪▪ Stats

pass SyncWorkspace (0s)
skip SetupSystemTool (skipped)
pass SyncSystemProject(project2) (0s)
pass SyncSystemProject(project1) (0s)
pass RunTask(project2:install-tool) (cached)
pass RunTask(project1:create-dir) (cached)
pass RunTask(project1:create-other-file) (cached)


Actions: 6 completed (3 cached), 1 skipped
   Time: 19ms
```

If you're testing this locally, you'll need to set `CI=true` since it actually checks if you're in a CI environment.

**Comment by milesj** [comment]
@mrobinsn `moon ci` is only meant to run in CI environments, it doesn't setup everything to _be_ a CI environment.

I'll look into forcing this CI check somehow, but it will work just fine in an actual CI environment since all of the providers set the environment variables.

**Comment by milesj** [comment]
Ok fixed in 1.16.1

**Comment by mrobinsn** [comment]
@milesj Thank you!
